### PR TITLE
fix: refactor events_file into Arg

### LINKED ISSUE #544: `soroban events` subcommand should accept a `--network` flag
### What version are you using?

Latest main

### What did you do?

`soroban events -h`

### What did you expect to see?

A `--network` flag option.

### What did you see instead?

That option was not present. It probably got missed when we were adding the `--network` flag.

### PR DESCRIPTION
**Pull request by willemneal**
### What

Fixes #544, adds new `events_file.rs`, and refactors some functions into instance methods in some `rpc` types.

### Why

This allows invoke and events commands to share the implementation and improves the code organization the code a bit.

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: cmd/soroban-cli/src/rpc/mod.rs [hunk 1]
```diff
@@ -150,6 +237,104 @@ pub struct Event {
     pub value: EventValue,
 }
 
+impl Display for Event {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```
Comment (paulbellamy) [inline cmd/soroban-cli/src/rpc/mod.rs:241]: Nice refactor moving this here.

**Review by paulbellamy — CHANGES_REQUESTED** [review]

File: cmd/soroban-cli/src/commands/events.rs [hunk 2]
```diff
@@ -202,11 +194,11 @@ impl Cmd {
             }
         }
 
-        let response = match (self.rpc_url.as_ref(), self.events_file.as_ref()) {
-            (Some(rpc_url), _) => self.run_against_rpc_server(rpc_url, start).await,
-            (_, Some(path)) => self.run_in_sandbox(path, start),
-            // should never happen because of required_unless_present flags
-            _ => return Err(Error::MissingTarget),
+        let response = if self.network.is_no_network() {
```
[the remark this answers is not in the document]
Comment (willemneal) [reply cmd/soroban-cli/src/commands/events.rs:197]: Yep and the tests still pass...

File: cmd/soroban-cli/src/commands/events.rs [hunk 3]
```diff
@@ -43,7 +43,7 @@ pub struct Cmd {
     /// the command-line today, though that set can have multiple IDs/topics.
     #[arg(
         long = "id",
-        num_args = 1..=5,
+        num_args = 1..=6,
```
Comment (paulbellamy) [inline cmd/soroban-cli/src/commands/events.rs:46]: ?

Comment (willemneal) [reply cmd/soroban-cli/src/commands/events.rs:46]: The docs above say the max is 5 and ..6 means at most 5. Unless the comment is wrong

**Review by paulbellamy — APPROVED** [review]

**Pull request merged**